src/System.Private.Xml/src/System/Xml/Serialization/XmlReflectionImporter.cs

```diff
@@ -1833,6 +1833,33 @@ private void ImportAccessorMapping(MemberMapping accessor, FieldModel model, Xml
                     accessor.Xmlns = new XmlnsAccessor();
                     accessor.Ignore = true;
                 }
+                else if(a.XmlAttribute != null && accessor.TypeDesc.IsOptionalValue)
+                {
```

Comment:
You copied this code from around line 1742. That code has the `a.XmlAttribute != null` check in an inner if and there's some other code which runs if it is null. Would we want to run that code in the case of a.XmlAttribute == null when IsOptionalValue is true? I think we might be able to modify the code around line 1738 to be this instead:
```csharp
var typeDesc = accessor.TypeDesc;
if (typeDesc.IsOptional)
{
    typeDesc = typeDesc.BaseTypeDesc;
}

if (typeDesc.IsPrimitive || typeDesc.IsEnum)
{
    ....
```
Your current change doesn't enforce the optional value being a primitive or an Enum. I suspect this is caught further up the call stack, but checking for this is done further up so there might a case when it isn't.